src/clj/rems/application/model.clj

```diff
@@ -682,10 +672,18 @@
              form/enrich-form-field-visible
              application))
 
+(defn- get-disable-command [application command]
+  (cond
+    (map? command) (for [[cmd states] command
+                         :when (contains? (set states) (:application/state application))]
+                     cmd)
+    :else [command]))
+
 (defn- enrich-disable-commands [application get-config]
```

Comment:
There's something a bit unclear with how this was setup a long time ago. There exists `application-permissions-for-workflow-view` which probably doesn't have the full (injected) application state to do the decisions there so this `enrich-disable-commands` is added here. I guess not much can be done here for this change, but some refactoring opportunity for later.

Reply:
Since full workflow body is not available here, we cannot use dynamic workflow for changing user permissions. `(application-view)` itself declares to be pure function, so it seems that design would need to be updated.

(Change which this comment originally refers to has been removed, but comment is still relevant)